### PR TITLE
Made the temperature gun less shit.

### PR DESCRIPTION
**Pull request by Buggy123**
It has significantly lower and higher temperature ranges, self charges, and no longer costs silver to make. As is, the temperature gun is a joke, maybe 1 person has ever died to it, and no one ever, ever uses it.

I'd set the freeze beam temperature even lower but I dunno if negative temps would break shit.

Also adds a "neutral" mode, which will instantly set someone's temperature to normal even if they're on fire/in space.

### CODE REVIEW

**Comment by duncathan** [comment]
where were you when someone died to a temp gun?

File: code/modules/projectiles/projectile/special.dm [hunk 1]
```diff
@@ -49,7 +49,7 @@
 	damage_type = BURN
 	nodamage = 1
 	flag = "energy"
-	var/temperature = 100
+	var/temperature = 0
```
Comment (duncathan) [inline code/modules/projectiles/projectile/special.dm]: NOPE, make it TCMB, not 0

**Comment by Buggy123** [comment]
Pay no attention to accidental closings/reopenings.

**Comment by newfren** [comment]
https://github.com/tgstation/-tg-station/issues/11017
https://github.com/tgstation/-tg-station/issues/11017
https://github.com/tgstation/-tg-station/issues/11017

**Comment by Buggy123** [comment]
He didn't even die to it, it just slowed him enough for someone to stun him.

...Do i need to buff this even more?

**Comment by AdamElTablawy** [comment]
Having a mode that instantly sets temperature to normal will be a pretty severe nerf for fire/plasmafire.

And again, this is just gonna encourage doctors running around with tempguns freezebeaming people and shooting them with syringes of cyroxadone. Might as well replace cyro with a room of tempguns and cyrox pills.

Them being available at roundstart w/o silver is gonna be a pretty heavy nerf to ops, too, and shadowlings, because if you shoot an op with freeze he's gonna be a popsicle in his hardsuit and suffer EXTREME slowdown. Same goes for sec in armor or anyone wearing a hardsuit. Same goes for shadowlings shot with the heat beam-roasting alive won't be fun.

You're also gonna have tag teams of scientists running through space to the armory blasting each other with tempguns to ignore space.

I dunno, the silver requirement with the buffs at least prevents R&D from vomiting 82 of these things out and it being the new, most loved weapon of revs for all the slowdown of disablers and all the damage of lasers for security in armor or hardsuits.

**Comment by Jazaen** [comment]
Are you afraid that the FUCKING TEMP GUN will get overpowered? Besides, why use tempguns to spacewalk when you can use hardsuits? And tempguns don't work on pressure damage.
But yeah, heat beam max temperature is a bit too high. Maybe make it more effective on simplemobs, so we have a dedicated anti-xeno weapon that won't fuck entire hallways (flamethrower)

**Comment by AdamElTablawy** [comment]
Yeah, i'm afraid the fucking tempgun will be too powerful.

Because one zap with the tempgun and an op in a hardsuit is moving at the speed of dead snail and freezing his dick off.

**Comment by Buggy123** [comment]
You can already use the temp gun to spacewalk without freezing, the current hot setting is barely above the heat damage threshold, so you can use it without taking more than the tinniest bit of damage before space cools you down.

**Comment by Lumipharon** [comment]
tempgun is mainly shit because the standard spessman normalises their temp at an utterly absurd rate.

That's fine for making space exposure not utter cancer, but it makes this weapon bum fuck useless, unless the actual mechanics of how it works are changed.

**Comment by GunHog** [comment]
What if the beam _raised_ or _lowered_ the target's temperature instead of hard-setting it? That way, 2-3 shots would not hurt a human at all, but taking several shots at once begins to get dangerous. Damage and temperature related slowdowns will make it easier to sustain fire.

**Comment by Buggy123** [comment]
I feel like that would be too much of a nerf again GunHog, that means you have to get 2-3+ shots in to start doing anything, when that anything is that you take slow burn damage and slow down.

**Comment by GunHog** [comment]
You could make it a bit more potent, then.

**Comment by Fox-McCloud** [comment]
I recommend looking into VG's temperature gun---which allows you to set a temperature on a scale---if you emag it, that temperature range increases.

At really high temps it ignites mobs---it's pretty cool and very useful for xenobio (not that the latter will matter on TG though, since Xenobio is uber easy-mode).

**Comment by KorPhaeron** [comment]
>are you afraid the fucking tempgun will be overpowered?

Tempgun actually used to be the strongest gun you could get short of adminspawn

**Comment by Jazaen** [comment]
I retract my point then.

File: code/modules/research/designs/weapon_designs.dm [hunk 2]
```diff
@@ -101,7 +101,7 @@
 	id = "temp_gun"
 	req_tech = list("combat" = 3, "materials" = 4, "powerstorage" = 3, "magnets" = 2)
 	build_type = PROTOLATHE
-	materials = list(MAT_METAL = 5000, MAT_GLASS = 500, MAT_SILVER = 3000)
```
Comment (KorPhaeron) [inline code/modules/research/designs/weapon_designs.dm]: Please don't remove the silver requirements

**Comment by Fox-McCloud** [comment]
https://github.com/d3athrow/vgstation13/pull/1681

Here's VG's temp gun implementation; obviously how it's handled would be a bit different on TG, but you could still pass the necessary var through ammo casings for the same effect (much like the portal gun does).

**Comment by Buggy123** [comment]
Closing this and making a better PR since i'm porting /vg/'s gun.

**Pull request closed without merging**